src/main/java/org/terasology/climateConditions/HyperthermiaSystem.java

```diff
@@ -124,12 +69,44 @@ private void applyWeakening(EntityRef player, HealthComponent health, Hypertherm
 
 
     /**
-     *  Reverts the player weakening by restoring the maxHealth and regeneration of the player to the original value.
+     * Reverts the player weakening by restoring the maxHealth and regeneration of the player to the original value.
      */
     private void revertWeakening(EntityRef player, HealthComponent health, HyperthermiaComponent hyperthermia) {
         player.send(new ChangeMaxHealthEvent(player.getParentPrefab().getComponent(HealthComponent.class).maxHealth));
         player.send(new ActivateRegenEvent());
         health.regenRate /= hyperthermia.regenMultiplier;
         player.saveComponent(health);
     }
+
+    @ReceiveEvent
+    public void hyperthermiaLevelChanged(HyperthermiaLevelChangedEvent event, EntityRef player,
+                                         HyperthermiaComponent hyperthermia, HealthComponent health) {
+        modifyHyperthermiaMultipliers(player, hyperthermia, event.getNewLevel());
+        //Adding New Effects when Hypothermia Level Increased.
+        if (event.getNewLevel() == 3) {
+            applyWeakening(player, health, hyperthermia);
+        } else if (event.getOldLevel() == 3) {
+            revertWeakening(player, health, hyperthermia);
+        }
+    }
+
+    private void modifyHyperthermiaMultipliers(EntityRef player, HyperthermiaComponent hyperthermia, int level) {
+        switch (level) {
```

Comment:
There is a `default` case missing here - what happens if the level is neither 1, 2, or 3? I may just be fine to log a warning about an "unexpected level" or something like that.